server/src/main/java/org/elasticsearch/index/store/LuceneFilesExtensions.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.store;
+
+import org.elasticsearch.common.Nullable;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public enum LuceneFilesExtensions {
+
+    CFE("cfe", "Compound Files Entries", true, false),
+    // Compound files are tricky because they store all the information for the segment. Benchmarks
+    // suggested that not mapping them hurts performance.
+    CFS("cfs", "Compound Files", false, true),
+    CMP("cmp", "Completion Index", true, false),
+    DII("dii", "Points Index", false, false),
+    // dim files only apply up to lucene 8.x indices. It can be removed once we are in lucene 10
+    DIM("dim", "Points", false, true),
+    // MMapDirectory has special logic to read long[] arrays in little-endian order that helps speed
+    // up the decoding of postings. The same logic applies to positions (.pos) of offsets (.pay) but we
+    // are not mmaping them as queries that leverage positions are more costly and the decoding of postings
+    // tends to be less a bottleneck.
+    DOC("doc", "Frequencies", false, true),
+    // Doc values are typically performance-sensitive and hot in the page
+    // cache, so we use mmap, which provides better performance.
+    DVD("dvd", "DocValues", false, true),
+    DVM("dvm", "DocValues Metadata", true, false),
+    FDM("fdm", "Field Metadata", true, false),
+    FDT("fdt", "Field Data", false, false),
+    FDX("fdx", "Field Index", false, false),
+    FNM("fnm", "Fields", true, false),
+    // old extension
+    KDD("kdd", "Points", false, true),
+    // old extension
+    KDI("kdi", "Points Index", false, true),
+    // Lucene 8.6 point format metadata file
+    KDM("kdm", "Points Metadata", true, false),
+    LIV("liv", "Live Documents", false, false),
```

Comment:
With the current implementation of live docs, they are fully read when opening an index, should we treat them as metadata?

Reply:
IIRC we did not flag `liv` files as metadata since we were expecting most indices to use soft-deletes and also because in my mind liv files can become large (?)

Reply:
`.liv` files can indeed be quite large, but so can be `.cmp` files.